### PR TITLE
`azurerm_attestation_provider` - add AzureVM and SEV-SNP attestation types

### LINKED ISSUE #21918: Support for trusted launch and SEV-SNP attestation policy management
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Support for managing attestation policies has been added by #20972 , unfortunately the [Azure documentation on the available TEE types ](https://learn.microsoft.com/en-us/azure/attestation/quickstart-azure-cli#policy-management) is outdated and does not mention the attestation types `AzureVM` and `SEV-SNP`

As seen in the Azure portal:
attestation types shown in the azure portal

The Terraform provider should offer support for these two attestation types, respectively.

### New or Affected Resource(s)/Data Source(s)

azurerm_attestation_provider

### Potential Terraform Configuration

```hcl
resource "azurerm_attestation_provider" "example" {
  name                = "exampleprovider"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  azure_vm_policy_base64 = "..."
}

resource "azurerm_attestation_provider" "example" {
  name                = "exampleprovider"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  sev_snp_policy_base64 = "..."
}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by msanft**
### Context
- As stated in #21918, the Terraform provider (nor the SDK) currently support setting attestation policies for the `AzureVM` (Trusted launch) and `SEV-SNP` attestation types.

### Proposed Changes
- Add the `sev_snp_policy_base64` and `azure_vm_policy_base64` fields to the `azurerm_attestation_provider` resource.
- TODO: For now, I edited the `enums.go` file in the SDK manually to test the changes. I assume I would need to get a PR to the SDK upstreamed to get these changes in?

### Related Issue
- Closes #21918

### CODE REVIEW

**Comment by stephybun** [comment]
Thanks for this PR @msanft.

As you've already identified, we cannot accept changes to vendored files and an upstream change would need to happen before this could be merged. 

The SDK used here is generated from the Azure REST API spec published in [this repo](https://github.com/Azure/azure-rest-api-specs). Unfortunately the latest dataplane version for attestation is still missing this value so you will need to raise an issue over on that repo to have it added to the spec before we can regenerate the SDK and propagate those changes into the provider to expose this value.

Since this isn't something we can currently support I'm going to close this issue for the time being, but once this value has been added to the spec we can look into reopening this and adding it.

Thanks!

**Review by stephybun — CHANGES_REQUESTED** [review]
Apologies just saw that one of the enums is available - if you can remove the one that isn't available in the SDK from this PR then we can take look over this and get it merged

**Comment by msanft** [comment]
> Apologies just saw that one of the enums is available - if you can remove the one that isn't available in the SDK from this PR then we can take look over this and get it merged

I'll edit that so we can have that merged. Thanks a lot!

For the `AzureVM` type, I will try to get that incorporated into the spec and open a separate PR once it's available upstream.

File: vendor/github.com/tombuildsstuff/kermit/sdk/attestation/2022-08-01/attestation/enums.go [hunk 1]
```diff
@@ -66,9 +66,13 @@ const (
...
+	// TODO: Upstream and replace before merge, for testing purposes only
+	// TypeAzureVM is the AzureVM attestation type.
+	TypeAzureVM Type = "AzureGuest"
 )
 
 // PossibleTypeValues returns an array of possible values for the Type const type.
 func PossibleTypeValues() []Type {
-	return []Type{TypeOpenEnclave, TypeSevSnpVM, TypeSgxEnclave, TypeTpm}
+	// TODO: Upstream and replace before merge, for testing purposes only
+	return []Type{TypeOpenEnclave, TypeSevSnpVM, TypeSgxEnclave, TypeTpm, TypeAzureVM}
```
Comment (stephybun) [inline vendor/github.com/tombuildsstuff/kermit/sdk/attestation/2022-08-01/attestation/enums.go:77]: These changes still need to be removed

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @msanft.

The change made to the vendored SDK needs to be reverted. We also have a test failure

```
------- Stdout: -------
=== RUN   TestAccAttestationProvider_withPolicy
=== PAUSE TestAccAttestationProvider_withPolicy
=== CONT  TestAccAttestationProvider_withPolicy
    testcase.go:113: Step 1/2 error: Error running apply: exit status 1
        Error: updating value for `tpm_policy_base64`: attestation.PolicyClient#Set: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="Bad request" Message="Tpm attestation is not supported in the 'UKSouth' region."
          with azurerm_attestation_provider.test,
          on terraform_plugin_test.tf line 24, in resource "azurerm_attestation_provider" "test":
          24: resource "azurerm_attestation_provider" "test" {
--- FAIL: TestAccAttestationProvider_withPolicy (514.40s)
FAIL
```

**Comment by msanft** [comment]
> Thanks @msanft.
> 
> The change made to the vendored SDK needs to be reverted. We also have a test failure
> 
> ```
> ------- Stdout: -------
> === RUN   TestAccAttestationProvider_withPolicy
> === PAUSE TestAccAttestationProvider_withPolicy
> === CONT  TestAccAttestationProvider_withPolicy
>     testcase.go:113: Step 1/2 error: Error running apply: exit status 1
>         Error: updating value for `tpm_policy_base64`: attestation.PolicyClient#Set: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="Bad request" Message="Tpm attestation is not supported in the 'UKSouth' region."
>           with azurerm_attestation_provider.test,
>           on terraform_plugin_test.tf line 24, in resource "azurerm_attestation_provider" "test":
>           24: resource "azurerm_attestation_provider" "test" {
> --- FAIL: TestAccAttestationProvider_withPolicy (514.40s)
> FAIL
> ```

Hey @stephybun ,
Sorry for my negligence here, I reverted the changes to the vendored SDK.

re the failed test; Unfortunately, I cannot replicate the failure locally. Judging from the error message, I think we could either switch the region used to e.g. `westus`, which should have all attestation types, or simply not test the unavailable attestation types in `uksouth`. I think testing all types would of course be the preferred way, but if you're constrained with using UK-South, I can also remove the unavailable attestation types here.

**Pull request merged**